src/app/util/attribute-storage.cpp

```diff
@@ -21,13 +21,21 @@
 #include <app/reporting/reporting.h>
 #include <app/util/af.h>
 #include <app/util/attribute-storage.h>
+#include <app/util/generic-callbacks.h>
 #include <lib/support/CodeUtils.h>
 #include <lib/support/logging/CHIPLogging.h>
 #include <platform/LockTracker.h>
 
-#include <app-common/zap-generated/attribute-type.h>
+// Attribute storage depends on knowing the current layout/setup of attributes
```

Comment:
@bzbarsky-apple @vivien-apple @selissia  - I think the logic here is in error, but unsure if fixing codegen is right.

The problem I have is:
  - endpoint_config.h creates somthing that depends on callbacks
  - endpoint_config.h depends on app-specific zap file (i.e. different between say lock and light)
  - here we include a app-common that seems to probably include defines of all possible callbacks and this is NOT app specific. This couples a "codegen all" with "codegen app"
  
I think at some point we should have:
  - codegen callbacks.h on a per-app basis (no more common callback.h)
  - remove app-common callback.h and use the codegen one
 
Is my understanding correct?

Reply:
> This couples a "codegen all" with "codegen app"

"Codegen app" should be able to depend on "codegen all", right?

Reply:
I think we have two categories:
  - controllers/client libraries (e.g. darwin framework, java, chip-tool, others) that want to have access to every possible client interaction and if not used, those will be compiled-out at link time
  - devices, which I called "apps" above because the app-codegen. I believe these should declare what they use at least server side.

The issue I have with the current design:
  - callbacks.h claims all possible `*Init` exists (or at least forward declares it)
  - endpoint_config.h will only actually invoke a small subset of this by design (only actually app enabled clusters)

There is a mismatch between callbacks.h list and actual cpp implementation, which I believe should be fixed. I am ok with saying 'codegen all' is the client side and everything should be able to access any client stuff. However I would also claim that for server-side specifically, we should only depend on server codegen data.

Reply:
Servers totally rely on "codegen all" bits right now (e.g. cluster-objects.cpp) and rely on dead-stripping to remove the bits they are not using.